binarapps/expo-ts-template
# FEAT: init detox and eas config for e2e tests

# Description

This PR introduces setup and initial e2e test (currently it tests signing in) on detox.

~Fixes # (issue)~

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

## iOS
On iOS simulator creates via script so to test it, you should only type:
- `yarn test:e2e:ios`

## Android
On Android creating an emulator via script will be provided in the future, so you have to create in Android Studio emulator that  matches name with the name in config (`pixel_4`) and:
- `yarn test:e2e:android:emu`

To test on the attached device
- `yarn test:e2e:android:att`

**Unfortunately Android provides flaky tests so check only if the emulator opens via script, installs the app, and runs the test (no matter if successful or not).**

Also - android has issues with getting the views by id (which causes flaky tests) - detox team in their documentation proposes to use `waitFor` with `withTimeout` to treat it as a workaround. But - as it is in the documentation I think that it might be the final approach.
If tests are failing for you, you can deal with timeout in `withTimeout` function - increasing that might result in a successful test.

# Screenshot(s)

## Android
https://vimeo.com/861140200?share=copy

## iOS
https://vimeo.com/861142368?share=copy

**Test Configuration**:

- Firmware version: 
  OS: macOS 13.4.1
  Node: version: 16.16.0
  Yarn: version: 1.22.19
  npm: version: 8.11.0
  Watchman: version: 2023.08.14.00
  CocoaPods: version: 1.12.1
- Hardware: 
      CPU: (8) arm64 Apple M1
      Memory: 349.56 MB / 16.00 GB
- Toolchain:
- SDK:  
  iOS SDK:
     Platforms:
      - DriverKit 22.4
      - iOS 16.4
      - macOS 13.3
      - tvOS 16.4
      - watchOS 9.4
     Android SDK: Not Found
  IDEs:
      Android Studio: 2021.1 AI-211.7628.21.2111.8309675
      Xcode:
          version: 14.3.1/14E300c
          path: /usr/bin/xcodebuild

# Checklist:

- [x] Add correct label to your pull request
